cdap-app-fabric/src/main/java/io/cdap/cdap/internal/tethering/TetheringServerHandler.java

```diff
@@ -222,4 +227,51 @@ private void updateTetherStatus(HttpResponder responder, String peer, TetheringS
     }
     responder.sendStatus(HttpResponseStatus.OK);
   }
+
+  /**
+   * Processes the list of tethering program updates received from the client and updates the program statuses
+   */
+  private void processProgramUpdates(String content) {
+    if (content.isEmpty()) {
+      return;
+    }
+    TetheringProgramUpdate programUpdate = GSON.fromJson(content, TetheringProgramUpdate.class);
+    for (TetheringProgramUpdate.TetheringProgramUpdateRequest request : programUpdate.getUpdateRequests()) {
+      ProgramRunId receivedProgram = GSON.fromJson(request.getProgramRunId(), ProgramRunId.class);
+      ProgramRunId programRunId = new ProgramRunId(NamespaceId.SYSTEM.getNamespace(), receivedProgram.getApplication(),
+                                                   receivedProgram.getType(), receivedProgram.getProgram(),
+                                                   receivedProgram.getRun());
+      ProgramRunStatus programRunStatus = ProgramRunStatus.valueOf(request.getProgramStatus());
+      switch (programRunStatus) {
+        case RUNNING:
+          programStateWriter.running(programRunId, null);
+          break;
+        case STOPPING:
+          programStateWriter.stop(programRunId, 10);
+          break;
+        case COMPLETED:
+          programStateWriter.completed(programRunId);
+          break;
+        case KILLED:
+          programStateWriter.killed(programRunId);
+          break;
+        case FAILED:
+          programStateWriter.error(programRunId, null);
+          break;
+        case SUSPENDED:
+          programStateWriter.suspend(programRunId);
+          break;
+        case RESUMING:
+          programStateWriter.resume(programRunId);
+          break;
+        case REJECTED:
+          programStateWriter.reject(programRunId, null, null, null, null);
```

Comment:
Is it ok to set some of these params as nullable (if yes, make the params nullable in ProgramStateWriter)? Wondering if we should pass this info in the TetheringProgramUpdateRequest